draft-ietf-taps-arch.md

```diff
@@ -212,7 +212,7 @@ Allowing applications to interact with messages is backwards-compatible with exi
 
 Sockets, for protocols like TCP, are generally limited to connecting to a single address over a single interface. They also present a single stream to the application. Software layers built upon sockets often propagate this limitation of a single-address single-stream model. The Transport Services architecture is designed to handle multiple candidate endpoints, protocols, and paths; and support multipath and multistreaming protocols.
 
-Transport Services implementations are meant to be flexible at connection establishment time, considering many different options and trying to select the most optimal combinations ({{gathering}} and {{racing}}). This requires applications to provide higher-level endpoints than IP addresses, such as hostnames and URLs, which are used by a Transport Services implementation for resolution, path selection, and racing.
+Transport Services implementations are meant to be flexible at connection establishment time, considering many different options and trying to select the most optimal combinations ({{gathering}} and {{racing}}). This requires applications to provide higher-level endpoints than IP addresses, such as hostnames and URLs, which are used by a Transport Services implementation for resolution, path selection, and racing. Transport services implementations can further implement fallback mechanisms if connection establishment of one protocol fails or performance is detected to be unsatisfactory.
 
```

Comment:
I find the "can further implement" a bit confusing as it sound like it is an additional thing, but I guess it is a clarification as racing gives you fallback? So maybe "In this way, Transport Services implementations can provide..." or something similar

Reply:
I see both options: Using racing in first place, but having an alt-srv like mechanism to discover more candidates or use transport level redirects.

Reply:
Do I need to change something now or not?

Reply:
I still think something needs to change. Indicating that "fallback mechanisms if connection establishment of one protocol fails" is something different from racing is not correct I think and not consistent with how racing is described elsewhere. But the sentence describes two things. The first to me is a clarification of what racing provides. The second "fallback if performance is detected to be unsatisfactory" is different and could perhaps be part of the general description that is given before, but could also be an add on. Perhaps just split the parts up?